internal/analysis/analysis.go

```diff
@@ -3,6 +3,7 @@ package analysis
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/ossf/package-analysis/internal/pkgecosystem"
```

Comment:
I was trying to keep this dependency out of this library and keep it unaware of the concept of "phases".

I think it might be clearer to break this work up into multiple steps:
1) move internal/analysis -> internal/dynamicanalysis
2) rename "runPhases" to "dynamicPhases"
3) introduce a new library internal/analysis that pulls together the common code from `worker` and `analyze`.

Reply:
Yes, there's no real reason this function has to be in this package. I'm fine to move it elsewhere.
Happy to break things up into multiple steps